agent/rpc/peering/service.go

```diff
@@ -409,8 +408,15 @@ func (s *Server) validatePeeringInPartition(remotePeerID, partition string) erro
 		return fmt.Errorf("cannot read peering by ID: %w", err)
 	}
 
-	if peering != nil && peering.Partition == partition {
-		return fmt.Errorf("cannot create a peering within the same partition (ENT) or cluster (OSS)")
+	if peering != nil {
+		stateStorePart := peering.Partition
+		if stateStorePart == "" {
+			stateStorePart = "default"
+		}
+
+		if stateStorePart == partition {
+			return fmt.Errorf("cannot create a peering within the same partition (ENT) or cluster (OSS)")
+		}
```

Comment:
Turns out `peering.PartitionOrDefault()` is a thing.
```suggestion
	if peering != nil {
		if peering.PartitionOrDefault() == partition {
			return fmt.Errorf("cannot create a peering within the same partition (ENT) or cluster (OSS)")
		}
```

Reply:
We should use `acl.EqualPartitions` which treats `"" == "default"`